weaviate/collections/collection/sync.py

```diff
@@ -63,58 +66,83 @@ def __init__(
         consistency_level: Optional[ConsistencyLevel] = None,
         tenant: Optional[str] = None,
         properties: Optional[Type[Properties]] = None,
         references: Optional[Type[References]] = None,
     ) -> None:
-        super().__init__(connection, name, validate_arguments)
+        self._connection = connection
+        self.name = _capitalize_first_letter(name)
+        self.__cluster = _Cluster(connection)
+        self._validate_arguments = validate_arguments
+        self.__tenant = tenant
+        self.__consistency_level = consistency_level
+        self.__properties = properties
+        self.__references = references
+        self.__event_loop = event_loop
+
+        config = _ConfigCollection(
+            connection=connection,
+            name=name,
+            tenant=tenant,
+        )
 
         self.aggregate = _AggregateCollection(
-            self._connection, self.name, consistency_level, tenant
+            connection=self._connection,
+            name=self.name,
+            consistency_level=consistency_level,
+            tenant=tenant,
         )
         """This namespace includes all the querying methods available to you when using Weaviate's standard aggregation capabilities."""
-        self.config = _ConfigCollection(self._connection, self.name, tenant)
-        """This namespace includes all the CRUD methods available to you when modifying the configuration of the collection in Weaviate."""
+        self.backup = _CollectionBackup(
+            connection=self._connection,
+            name=self.name,
+        )
+        """This namespace includes all the backup methods available to you when backing up a collection in Weaviate."""
         self.batch = _BatchCollectionWrapper[Properties](
-            connection, consistency_level, self.name, tenant, self.config
+            event_loop,
+            self._connection,
+            self.__consistency_level,
+            self.name,
+            self.__tenant,
+            config,
         )
         """This namespace contains all the functionality to upload data in batches to Weaviate for this specific collection."""
+        self.config = config
+        """This namespace includes all the CRUD methods available to you when modifying the configuration of the collection in Weaviate."""
         self.data = _DataCollection[Properties](
-            connection, self.name, consistency_level, tenant, validate_arguments, properties
+            connection, self.name, consistency_level, tenant, validate_arguments
         )
         """This namespace includes all the CUD methods available to you when modifying the data of the collection in Weaviate."""
-        self.generate = _GenerateCollection(
-            connection,
-            self.name,
-            consistency_level,
-            tenant,
-            properties,
-            references,
-            validate_arguments,
+        self.generate = _GenerateCollection[Properties, References](
```

Comment:
there are also some unrelated changes in here?

Reply:
This is to fix a typing issue raised by `pyright` that was likely missed in previous versions